### PR TITLE
[READY] SNAKES!

### PR DESCRIPTION
**Pull request by Togopalis**
🆑 Togopal, Armhulenn, Naksu, Jordie & oranges
add: Snakes! They hunt vermin on the station! Curators have a natural phobia to them.
add: They can be purchased in cargo.
/🆑

Mice can be a problem in late round random events, and I figured releasing a bunch of snakes in maintenance would be a more long-term solution than constantly running around with a crowbar when you hear them.

Now with updated sprites!

(I apologize heavily for uploading this three times. I was unaware I was using an older fork, as this was my first attempt at modifying the code.)

From #34716 and #34706

### CODE REVIEW

**Comment by Hatterhat** [comment]
Yeah, OK, Togopal, I guess you could consider that an effort to add snakes.

**Comment by TheDreamweaver** [comment]
Snakes on a metal cage floating through the void of space? I wonder how this could go wrong...

**Comment by ExcessiveUseOfCobblestone** [comment]
STOP

**Comment by Togopalis** [comment]
YOU CAN'T TELL ME WHAT TO DO

**Comment by Magikarpet** [comment]
What do we release to stop the snakes, eagles or something?

**Comment by Togopalis** [comment]
Yes, but then how do you stop the eagles?

**Comment by Togopalis** [comment]
By the way FTR Arm is gonna teach me how to code Curators having a phobia to snakes

**Comment by DoctoreMad** [comment]
Are you using a sprite based on the alien worm intentionally

**Comment by Togopalis** [comment]
You gotta admit it's a pretty good base to use! Armhulenn was pretty smart on that decision. We may have a new sprite though depending on if @Supermichael777 wants to pitch in! He had a pretty cool unique sprite

**Comment by ayyourboibilly** [comment]
We do need more mobs and this is a small but neat one

**Comment by Togopalis** [comment]
Working some more on this!

Regular snakes that hunt mice will be able to be ordered from Cargo for 3000 points and exotic snakes (Ones with deadly toxin that can be extracted for nefarious deeds) will be available for 10000 from an emagged cargo console.

**Comment by Iatots** [comment]
But does it grow longer with every mouse it consumes?

**Comment by Togopalis** [comment]
I wish

File: code/modules/cargo/packs.dm [hunk 1]
```diff
@@ -1035,6 +1035,17 @@
...
+/datum/supply_pack/organic/critter/snake
+    name = "Exotic Snakes"
+    cost = 3000
+    contains = list(/mob/living/simple_animal/hostile/retaliate/poison/snake)
+    crate_name = "snake crate"
+
+/datum/supply_pack/organic/critter/snake/generate()
+    . = ..()
+    for(var/i in 1 to 3)
+        new /mob/living/simple_animal/hostile/retaliate/poison/snake(.)
```
Comment (Cyberboss) [inline code/modules/cargo/packs.dm]: Use the contains list

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/cargo/packs.dm (hunk 1 above)
Comment (Togopalis) [reply code/modules/cargo/packs.dm]: Ok dokey artichoke

Comment (Togopalis) [reply code/modules/cargo/packs.dm]: Nvm arm told me how to fix it

**Comment by Bawhoppen** [comment]
We have plenty of bad sprites, but nowadays we try and limit the actual bad new sprites we introduce. You're going to have to try and get a better one, rather than just recoloring the xeno larva.

**Comment by Iatots** [comment]
there is armhulen and the 77 guy both allegedly working on new sprites.

**Comment by Togopalis** [comment]
Armhulenns sprite was the alien larva, and its a lot more than a recolor. It has a lot of nice animations to it.

**Comment by PKPenguin321** [comment]
Just curious, how do phobias actually work? I think they were added when I was on my hiatus
Good PR by the way

**Comment by evsey9** [comment]
snakes should aggro at people when people get too close, or at least the exotic ones should

File: code/modules/mob/living/simple_animal/friendly/snake.dm [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	var/list/mice = list()
+	for (var/HM in .)
+		//Yum a tasty mouse
+		if(istype(HM, /mob/living/simple_animal/mouse))
+			mice += HM
+		if(isliving(HM))
+			living_mobs += HM
+
+	// if no tasty mice to chase, lets chase any living mob enemies in our vision range
+	var/enemies_in_range = list()
```
Comment (optimumtact) [inline code/modules/mob/living/simple_animal/friendly/snake.dm]: just remov this, it's not needed and I forgot toremove it

**Comment by optimumtact** [comment]
by the way it's oranges

**Pull request merged**